sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1002,12 +1002,13 @@ object EliminateSorts extends Rule[LogicalPlan] {
 
   private def isOrderIrrelevantAggs(aggs: Seq[NamedExpression]): Boolean = {
     def isOrderIrrelevantAggFunction(func: AggregateFunction): Boolean = func match {
-      case _: Sum => true
       case _: Min => true
       case _: Max => true
       case _: Count => true
-      case _: Average => true
-      case _: CentralMomentAgg => true
+      // Arithmetic operations for floating-point values are order-sensitive
+      // (they are not associative).
+      case _: Sum | _: Average | _: CentralMomentAgg =>
```

Comment:
Also you can write I think `_: Sum | Average | CentralMomentAgg`

Reply:
Is that a legal syntax? I tried to change though, I hit a compiler error.

Reply:
Not sure honestly, I tried a similar simpler example locally and it worked but maybe not here. Disregard.

Reply:
ok, thanks!